File: 2.programming_fundamentals_may_2023/4.2.functions_exercise/04.odd_and_even_sum.py
# You will receive a single number.
# You should write a function that returns the sum of all even and all odd digits in a given number.
# The result should be returned as a single string in the format:
# "Odd sum = {sum_of_odd_digits}, Even sum = {sum_of_even_digits}"
# Print the result of the function on the console.
#
# Input1: 1000435
# Output1: Odd sum = 9, Even sum = 4
#
# Input2: 3495892137259234
# Output2: Odd sum = 54, Even sum = 22

def odd_even_numbers(some_number):
    sum_of_odd_digits = 0
    sum_of_even_digits = 0
    for digit in some_number:
        if int(digit) % 2 == 0:
            sum_of_even_digits += int(digit)
        else:
            sum_of_odd_digits += int(digit)
    return sum_of_odd_digits, sum_of_even_digits

number = input()
odd_digits, even_digits = odd_even_numbers(number)
print(f'Odd sum = {odd_digits}, Even sum = {even_digits}')

File: 6.js_front_end_feb_2024/3.2.js_functions_and_statements_exercise/04.odd_and_even_sum.js
function solve(number) {
    let oddSum = 0;
    let evenSum = 0;
    let digits = number.toString().split('');

    for (let digit of digits) {
        if (digit % 2 === 0) {
            evenSum += Number(digit);
        } else {
            oddSum += Number(digit);
        }
    }

    console.log(`Odd sum = ${oddSum}, Even sum = ${evenSum}`);
}

solve(1000435);
solve(3495892137259234);
